client/__tests__/e2e/e2e.test.js

```diff
@@ -309,6 +309,58 @@ describe("ui elements don't error", () => {
       panCoords.end,
       false
     );
-    await page.evaluate(`window.scrollBy(0, 1000);`);
+    await page.evaluate("window.scrollBy(0, 1000);");
+  });
+});
+
+describe("centroid labels", () => {
+  test("labels are created", async () => {
+    await utils.clickOn("centroid-label-toggle");
+    const labels = Object.keys(data.categorical);
+    /* eslint-disable no-await-in-loop */
+    // Toggle colorby for each category and check to see if labels are generated
+    for (let i = 0, { length } = labels; i < length; i += 1) {
```

Comment:
nit: I think here you can use a [`for-of`](https://stackoverflow.com/questions/3010840/loop-through-an-array-in-javascript) statement and make things a little cleaner

Reply:
IIRC: `for of`'s don't work nicely with async/await

Reply:
they are also heinously slow.   We strongly dislike for..or in any situation where performance matters even a bit.

This code matches our current conventions.

Reply:
Ah! Ok, good to know.